Modules/Wishlist/Http/Controllers/WishlistController.php

```diff
@@ -20,63 +22,201 @@ public function __construct(WishlistService $wishlistService)
     }
 
     /**
+     * Add wishlist by logged user
+     *
      * @param Request $request
      * @return JsonResponse
      */
     public function add(Request $request): JsonResponse
     {
+        $user = auth()->user();
+        if ($user->cannot('create', Wishlist::class)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
         $response = $this->wishlistService->add($request);
+
         return response()->json($response);
     }
 
     /**
+     * Fetch wishlist by logged user
+     *
      * @param Request $request
      * @return JsonResponse
      */
     public function fetch(Request $request): JsonResponse
     {
+        $user = auth()->user();
+        if ($user->cannot('viewAny', Wishlist::class)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
         $response = $this->wishlistService->fetch($request);
+
+        return response()->json($response);
+    }
+
+    /**
+     * Delete wishlist by ID
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function delete(Request $request): JsonResponse
+    {
+
+        $user = auth()->user();
+        $wishList = Wishlist::where('id', $request->id)->first();
+        if ($user->cannot('delete', $wishList)) {
+            return response()->json([
+                'res' => false,
+                'msg' => 'User is not authorized !',
+                'data' => ""
+            ]);
+        }
+        $response = $this->wishlistService->delete($request);
+
         return response()->json($response);
     }
 
+    /**
+     * Fetch boards which created by logged user
+     *
+     * @param Request $request
+     * @return JsonResponse
+     */
+
     public function fetchBoards(Request $request): JsonResponse
```

Comment:
$request is unnecessary parameter, remove

Reply:
Done